### PR TITLE
KTOR-5753: Enable linuxArm64 builds

### PR DESCRIPTION
**Pull request by bcmedeiros**
**Subsystem**
Server and Client

**Motivation**
https://youtrack.jetbrains.com/issue/KTOR-5753

**Solution**
This is spike to enable linuxArm64 for ktor.

Caveats:

- ~This is using `kotlinx-coroutines` `1.7.0-RC`, so we should probably wait for the final GA release, and possibly also make this upgrade as a separate PR. Happy to do that if you guys think it's the best way~ done, https://github.com/ktorio/ktor/pull/3609. ✅ 
- ~This is using an un-published version of `kotlinx-html` containing https://github.com/Kotlin/kotlinx.html/pull/213, as the last published version does not support `linuxArm64`~. Using kotlinx.html 0.9.0  ✅
- ~I ended up copying a few files from `linuxX64` to the the new `linuxArm64`. Are we planning to introduce a common `linux` source set?~ Done, files are not copied anymore. ✅ 

Any feedback? Happy to keep helping with this as I need ktor server running on linux arm for one of my projects.

### CODE REVIEW

File: ktor-network/linuxArm64/src/io/ktor/network/selector/SelectUtilsLinux.kt
Comment (hfhbd) [inline ktor-network/linuxArm64/src/io/ktor/network/selector/SelectUtilsLinux.kt:1]: Instead copying, could you move this file into linuxMain?

File: gradle.properties [hunk 1]
```diff
@@ -32,8 +32,8 @@ kotlin.native.binary.memoryModel=experimental
 #kotlinx.atomicfu.enableJsIrTransformation=true
 
 kotlin_version=1.8.10
-coroutines_version=1.6.4
-atomicfu_version=0.20.0
+coroutines_version=1.7.0-RC
```
Comment (hfhbd) [inline gradle.properties:35]: 1.7.0 is released

**Comment by hfhbd** [comment]
You should be able to move many copied files into `linuxMain`.

File: ktor-network/linuxArm64/src/io/ktor/network/selector/SelectUtilsLinux.kt
Comment (bcmedeiros) [reply ktor-network/linuxArm64/src/io/ktor/network/selector/SelectUtilsLinux.kt:1]: I'll try that, I think I was trying `linux` before.

File: gradle.properties (hunk 1 above)
Comment (bcmedeiros) [reply gradle.properties:35]: Raised https://github.com/ktorio/ktor/pull/3609, which contains more things than the automatic one.

**Comment by bcmedeiros** [comment]
> You should be able to move many copied files into `linuxMain`.

@hfhbd  The project didn't have a `linux` common folder set up, but I created one and could just move the files to a common place, everything is compiling now. Let me know what you think.

**Comment by bcmedeiros** [comment]
> I am okay with this, but I am not member of Ktor. Especially with serialization 1.5.1, we could add the linuxArm64 target to more ktor modules.

Nice, I'm planning to keep this PR up-to-date until it's ready to be merged. There are two blocker PRs in the description, one for coroutines update in this repo, and another one for kotlinx-html, so I think we need some ktor/JB sponsor here.

**Comment by bcmedeiros** [comment]
@e5l this one is now ready to review.

**Review by e5l — APPROVED** [review]

**Comment by e5l** [comment]
@bcmedeiros could you rebase it on main?

**Comment by bcmedeiros** [comment]
@e5l I thought you'd squash merge it.
Do you want me to rebase it as as single commit? I can do that in about 30 min.

**Comment by e5l** [comment]
Thanks for the quick answer! No problem, I'll wait. The kotlinx.html is published, and I also want to run tests the rebased version

**Comment by bcmedeiros** [comment]
@e5l just pushed a single commit with the same net result, let me know if you need anything else :)

**Comment by bcmedeiros** [comment]
> The kotlinx.html is published, and I also want to run tests the rebased version
I was already using the new pusblished version of `kotlinx.html` before the rebase.

**Comment by bcmedeiros** [comment]
@e5l is any of that related to arm support? Should we split the `kotlinx-html` upgrade into a separate PR?

**Comment by hfhbd** [comment]
Well, kotlinx.html does not specify the jvmtoolchain, so it uses the jvm from TeamCity, which is Java 11, but ktor requires Java 8. Ideally, kotlinx.html (and ktor) should specify the jvmtoolchain explicitly.
Kotlinx.html still requires a new release.

**Comment by e5l** [comment]
@hfhbd, yep. WIP

**Comment by e5l** [comment]
@bcmedeiros, we can't. There was no arm version for 0.8.*. I will make new release soon

**Comment by e5l** [comment]
kotlinx.html `0.9.1` is out. Could you please update?

**Comment by bcmedeiros** [comment]
@e5l done :)

**Comment by e5l** [comment]
@bcmedeiros, Awesome Job! Thanks. Merged

**Comment by hfhbd** [comment]
https://repo1.maven.org/maven2/io/ktor/ktor-server-cio-linuxarm64/ 🎉

**Pull request merged**